extensions/amp-analytics/0.1/amp-analytics.js

```diff
@@ -258,7 +261,9 @@ export class AmpAnalytics extends AMP.BaseElement {
           this.user().error(TAG, 'Trigger should be an object: ', k);
           continue;
         }
-        if (!trigger['on'] || !trigger['request']) {
+        const hasRequestOrPostMessage = trigger['request'] ||
+            (trigger['parentPostMessage'] && this.isInabox_);
+        if (!trigger['on'] || !hasRequestOrPostMessage) {
           this.user().error(TAG, '"on" and "request" ' +
               'attributes are required for data to be collected.');
```

Comment:
This is now a somewhat misleading error message, since `request` is no longer required.  For example, if someone misspelled `parentPostMessage` they might think this was telling them they also needed to add `request`

Reply:
That's a good point. I refined the error message a bit in inabox and non inabox case.